docs/reference/search/retriever.asciidoc

```diff
@@ -371,6 +374,118 @@ GET movies/_search
 ----
 // TEST[skip:uses ELSER]
 
+[[rescorer-retriever]]
+==== Rescorer Retriever
+
+The `rescorer` retriever re-scores only the results produced by its child retriever.
+For the `standard` and `knn` retrievers, the `window_size` parameter specifies the number of documents examined per shard.
```

Comment:
Did we name this `window_size` to be in sync with the top-level `rescore` API? We have made a distinction on that and retrievers (where the top-level documents that a retriever operates upon are specified through `rank_window_size`), but I can see how that would make it more straightforward for users to move to the new framework. 

Either is fine though I guess :)

Reply:
Yep, I am re-using the rescorer builder parser to ease the integration. I like `window_size` better, what do you think of accepting both in the `CompoundRetrieverBuilder`? I can rename in the documentation to consistently refer to `window_size` while still accepting `rank_window_size`.

Reply:
Hmm I thinks its fine accepting either argument but I'm not so sure about the documentation changes tbh. This was renamed in https://github.com/elastic/elasticsearch/pull/106253, to differentiate at the time between rescore's window size (as one is applied on the shards and pagination/expected results could behave slightly differently). I don't have a strong preference however.

Reply:
I am also fine keeping the distinction. The `window_size` here refers to rescorer so it would be inconsistent to name it `rank_window_size` while the rescorer doc keeps the other name?

Reply:
yeah, ++ to keep it as `window_size` just for the rescorer retriever. (should we also accept rankWindowSize and/or properly handle the param references in `CompoundRetrieverBuilder#validate` ) ?